utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,254 @@
+            if s1 is None:
+                latest_files[ext] = hist
+                continue
+            (yr,month,day,time) = s1.group(1,2,3,4)
+            s2 = date_match.search(latest_files[ext])
+            (pyr,pmonth,pday,ptime) = s2.group(1,2,3,4)
+            if yr > pyr or (yr == pyr and month > pmonth) or \
+                    (yr == pyr and month == pmonth and day > pday) or \
+                    (yr == pyr and month == pmonth and day == pday and time > ptime):
+                latest_files[ext] = hist
+                logger.debug("ext %s hist %s %s"%(ext,hist,latest_files))
+        else:
+            latest_files[ext] = hist
+
+    for key in latest_files.keys():
+        histlist.append(latest_files[key])
+    return histlist
+
+
+def move(case, suffix):
+    """
+    Change the suffix for the most recent batch of hist files in a case. This can
+    allow you to temporarily "save" these files so they won't be blown away if you
+    re-run the case.
+
+    case - The case containing the files you want to save
+    suffix - The string suffix you want to add to saved files, this can be used to find them later.
+    """
+    rundir   = case.get_value("RUNDIR")
+    testcase = case.get_value("CASE")
+
+    # Loop over models
+    comments = "Moving hist files to suffix '%s'\n" % suffix
+    num_moved = 0
+    for model in _iter_model_file_substrs(case):
+        comments += "  Moving hist files for model '%s'\n" % model
+        test_hists = _get_latest_hist_files(testcase, model, rundir)
+        num_moved += len(test_hists)
+        for test_hist in test_hists:
+            new_file = "%s.%s" % (test_hist, suffix)
+            if os.path.exists(new_file):
+                os.remove(new_file)
+
+            comments += "    Copying '%s' to '%s'\n" % (test_hist, new_file)
+            shutil.copy(test_hist, new_file)
+
+    expect(num_moved > 0, "move failed: no hist files found in rundir '%s'" % rundir)
+
+    return comments
+
+def _compare_hists(case, from_dir1, from_dir2, suffix1="", suffix2=""):
+    if from_dir1 == from_dir2:
+        expect(suffix1 != suffix2, "Comparing files to themselves?")
+
+    testcase = case.get_value("CASE")
+    all_success = True
+    num_compared = 0
+    comments = "Comparing hists for case '%s' dir1='%s', suffix1='%s',  dir2='%s' suffix2='%s'\n" % \
+        (testcase, from_dir1, suffix1, from_dir2, suffix2)
+
+    for model in _iter_model_file_substrs(case):
+        comments += "  comparing model '%s'\n" % model
+        hists1 = _get_latest_hist_files(testcase, model, from_dir1, suffix1)
+        hists2 = _get_latest_hist_files(testcase, model, from_dir2, suffix2)
+        len_hist1 = len(hists1)
+        len_hist2 = len(hists2)
+        if len_hist1 == 0:
+            comments += " no hist files found for model %s\n"%model
+            continue
+        if len_hist1 != len_hist2:
+            comments += "    num hists does not match %d != %d\n" % (len_hist1, len_hist2)
```

Comment:
Can you elaborate? The sets of files will not match in many cases. For example, if suffix1 != suffx2, then it would be impossible for the sets of hist files to be identical unless you made some effort to normalize the names.